modules/proxy-certificates.adoc

```diff
@@ -0,0 +1,133 @@
+
+The `trustedCA` field should only be consumed by a proxy validator. The
+validator is responsible for reading the certificate bundle from required key
+`ca-bundle.crt` and copying it to a ConfigMap named `trusted-ca-bundle` in the
+`openshift-config-managed` namespace. The namespace for the ConfigMap referenced
+by `trustedCA` is `openshift-config`:
+
+[source,yaml]
+----
+apiVersion: v1
+	kind: ConfigMap
+ metadata:
+	 name: user-ca-bundle
+	 namespace: openshift-config
+	 data:
+	   ca-bundle.crt: |
+	     -----BEGIN CERTIFICATE-----
+	     Custom CA certificate bundle.
+	     -----END CERTIFICATE-----
+----
+
+== File path
+
+The user-provided trust bundle is represented as a ConfigMap. The ConfigMap is
+mounted into the filesystem of platform components that make egress HTTPS calls.
+Typically, Operators mount the ConfigMap to
+`/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`, but this not required by
+proxy and is at the Operator's discretion.
+
+If using the RHCOS trust bundle, place CA certificates in
+`/etc/pki/ca-trust/source/anchors`.
+
+See
+link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/using-shared-system-certificates_security-hardening[Using
+shared system certificates] in the Red Hat Enterprise Linux documentation for
+more information.
+
+== Expiration
+
+The user sets the expiration term of the user-provided trust bundle.
+
+The default expiration term is defined by the CA certificate itself. It is up to
+the CA administrator to configure this for the certificate before it can be used
+by {product-title} or RHCOS.
+
+[NOTE]
+====
+Red Hat does not monitor for when CAs expire. However, due to the long life of
+CAs, this is generally not an issue. However, you might need to periodically
+update the trust bundle.
+====
+
+== Services
+
+When proxy is enabled, all platform components that make egress HTTPS calls will
+use either:
+
+* the merged user/system trust bundle if `trustedCA` is specified, or
+* the RHCOS trust bundle if `trustedCA` is not specified.
+
+Any service that is running on the RHCOS node would be able to use the trust
+bundle of the node.
+
+== Customization
+
+Updating the user-provided trust bundle consists of either:
+
+* updating the PEM-encoded certificates in the ConfigMap referenced by
+`trustedCA,` or
+* creating a ConfigMap in the namespace `openshift-config` that contains the new
+trust bundle and updating `trustedCA` to reference the name of the new
+ConfigMap.
+
+The mechanism for writing CA certificates to the RHCOS trust bundle is exactly
+the same as writing any other file to RHCOS, which is done through the use of
+MachineConfigs. When the Machine Config Operator (MCO) applies the new
+MachineConfig that contains the new CA certificates, the node is rebooted.
+During the next boot, the service `coreos-update-ca-trust.service` runs on the
+RHCOS nodes, which automatically update the trust bundle with the new CA
+certificates. For example:
```

Comment:
@runcom per https://bugzilla.redhat.com/show_bug.cgi?id=1784201 the trust store of machines should also support updating the trust store of nodes based on the trusted ca injection process outlined [here](https://docs.google.com/document/d/1y0t0yEOSnKc4abxsjxEQjrFa1AP8iHcGyxlBpqGLO08/edit?usp=sharing). cc: @ahardin-rh